R/gq_design.R

```diff
@@ -0,0 +1,40 @@
+#' Design with Gaussian quadrature integration rule
+#'
+#'     gq_design simply creates a TwoStageDesign object with a Gaussian
+#'     quadrature numerical integration rule. If length(n2) = 1 and
+#'     length(c2) > 1 an object of class \code{GSDesign} is created.
+#'
+#' @param n1 stage-one sample size
+#' @param c1f early stopping for futility boundary
+#' @param c1e early stopping for efficacy boundary
+#' @param n2 second-stage sample size. vector for adaptive and
+#'           integer for group-sequential design
+#' @param c2 secnod-stage decision boundary
+#' @param order order (i.e. number of pivot points in the interior of [c1f, c1e])
+#'     of the Gaussian quadrature rule to use for integration
+#' @param ... futher arguments
+#'
+#' @export
+setGeneric("gq_design", function(n1, c1f, c1e, n2, c2, order = 5, ...)
+    standardGeneric("gq_design"))
+
+#' @describeIn gq_design Method to create design with Gaussian quadrature rule
+#'             and Legendre nodes
+#' @export
+setMethod("gq_design", signature("numeric"),
```

Comment:
Why are you only including the first argument in the signature? Is c1f arbitrary? No must also be numeric, right?